### PR TITLE
fix(design-system): dropdown closing after click on disabled option

### PR DESCRIPTION
**Pull request by mateus2033**
## Description of changes

* A validation has been implemented in the dropdown to identify whether a field is disabled or not.
   i - If activated, the Dropdown will be closed and its corresponding action will be called.
   ii - If disabled, the dropdown will not be closed and no action will be called. 
## GitHub issues resolved by this PR

- https://github.com/devopness/devopness/issues/1015#issue-2674862030

## Quality Assurance

- Once the changes in this PR are merged and deployed, success criteria is: 
   i - When you click on a disabled option, the dropdown should remain open and no action should be triggered.

## More info

N/A

### CODE REVIEW

**Comment by thlmenezes** [comment]
Hello @mateus2033 and thanks for opening your first PR.

Can you review the following:

- [ ] Update your PR description, filling out the fields in PR template
    - This helps us better understand what is being fixed and that you have tested it
- [ ] Review the CLA agreement
    - Read more @ https://github.com/devopness/devopness/pull/1102#issuecomment-2561360627
- [ ] Add a changeset to this PR
    - Read more @ https://github.com/devopness/devopness/pull/1102#issuecomment-2561360552
    - If you need help here, I can suggest one, just @ me

File: packages/ui/react/src/components/Primitives/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -313,16 +313,18 @@ const Dropdown = ({
...
-                          }
+                            if (option.onClick) {
+                              option.onClick()
+                            } else if (onSelect) {
+                              onSelect(option)
+                            }
 
-                          popupState.close()
+                            popupState.close()
+                          }
```
Comment (thlmenezes) [inline packages/ui/react/src/components/Primitives/Dropdown/Dropdown.tsx:327]: I believe we can avoid the extra indentation using an early return [1]

This helps whoever reads to follow the happy path by following the same indentation

Can you apply the suggestion below?

[1] https://medium.com/swlh/return-early-pattern-3d18a41bba8

This source was mentioned in multiple websites during my search for this pattern name, if you are interested
```suggestion
                          if(option.isDisabled) {
                            return
                          }

                          event.preventDefault()
                          event.stopPropagation()

                          if (option.onClick) {
                            option.onClick()
                          } else if (onSelect) {
                            onSelect(option)
                          }

                          popupState.close()
```

File: packages/ui/react/src/components/Primitives/Dropdown/Dropdown.tsx
Comment (thlmenezes) [inline packages/ui/react/src/components/Primitives/Dropdown/Dropdown.tsx:1]: I suggest updating the automated tests in src/components/Primitives/Dropdown/Dropdown.test.tsx

You can use a test case like the example below to validate the bug is fixed and avoid regressions in the future

```tsx
  it('click dropdown disabled option', () => {
    const onClick = vi.fn(() => null)

    render(
      <Dropdown
        id="test-dropdown"
        anchorType="button"
        label="Menu"
        options={[
          {
            label: 'Remove',
            isDisabled: true,
            badge: { icon: true, name: 'delete' },
            onClick,
          },
        ]}
      />
    )

    const subjectDropdown = screen.getByText('Menu')
    expect(subjectDropdown).toBeInTheDocument()

    expect(onClick).not.toHaveBeenCalled()

    act(() => {
      subjectDropdown.click()
    })

    const subjectRemove = screen.getByText('Remove')
    expect(subjectRemove).toBeInTheDocument()

    act(() => {
      subjectRemove.click()
    })

    expect(subjectRemove).toBeInTheDocument()
    expect(onClick).not.toHaveBeenCalled()
  })
```

**Review by thlmenezes — CHANGES_REQUESTED** [review]

**Comment by mateus2033** [comment]
> **Bot CLA Assistant Lite:** Obrigado 💖 por sua submissão! Pedimos que você assine nosso [Contrato de Licença de Colaborador](https://github.com/devopness/.github/blob/main/CLA.md) antes de aceitarmos sua contribuição. Você pode acompanhar o CLA adicionando um comentário abaixo usando este texto:
> 
> Li o documento do CLA e assino o CLA
> 
> Você pode reativar este bot comentando **recheck** neste Pull Request

Li o documento do CLA e assino o CLA

**Comment by mateus2033** [comment]
recheck

**Comment by mateus2033** [comment]
> Olá@mateus2033e obrigado por abrir seu primeiro RP.
> 
> Você pode revisar o seguinte:
> 
> * [ ] Atualize sua descrição de RP, preenchendo os campos no modelo de RP (Descrição das alterações, problemas do GitHub resolvidos por esta RP, ...)
>   
>   * Isso nos ajuda a entender melhor o que está sendo corrigido e como você testou as alterações
> * [ ] Revise o acordo CLA
>   
>   * Leia mais em [fix: correção para fechamento do menu suspenso  #1102 (comentário)](https://github.com/devopness/devopness/pull/1102#issuecomment-2561360627)
> * [ ] Adicionar um conjunto de alterações a este PR
>   
>   * Leia mais em [fix: correção para fechamento do menu suspenso  #1102 (comentário)](https://github.com/devopness/devopness/pull/1102#issuecomment-2561360552)
>   * Se precisar de ajuda aqui, posso sugerir uma, é só me mandar uma mensagem.

**Comment by mateus2033** [comment]
> [thlmenezes](/thlmenezes)

Hello thlmenezes, I would like an example of how to fill in the information.

**Comment by mateus2033** [comment]
I have read the CLA document and I hereby sign the CLA

File: packages/ui/react/src/components/Primitives/Dropdown/Dropdown.tsx (hunk 1 above)
Comment (mateus2033) [reply packages/ui/react/src/components/Primitives/Dropdown/Dropdown.tsx]: Suggestion added.

File: packages/ui/react/src/components/Primitives/Dropdown/Dropdown.tsx
Comment (mateus2033) [reply packages/ui/react/src/components/Primitives/Dropdown/Dropdown.tsx:1]: Suggestion added.

**Comment by thlmenezes** [comment]
@mateus2033 could you update this PR title? I believe we can be more specific

From:

```
fix: correction to dropdown closing
```

To:

```
fix(design-system): dropdown closing when clicking disabled option
```

- "fix" and "correction to" add the same meaning to the message
- add more context as to why the dropdown is closing and why it is a bug

File: .changeset/odd-nails-do.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@devopness/ui-react": minor
```
Comment (thlmenezes) [inline .changeset/odd-nails-do.md:2]:
```suggestion
"@devopness/ui-react": patch
```

> Given a version number MAJOR.MINOR.PATCH, increment the:
>
> 1. MAJOR version when you make incompatible API changes
> 1. MINOR version when you add functionality in a backward compatible manner
> 1. PATCH version when you make backward compatible bug fixes

Source: https://semver.org/

File: .changeset/odd-nails-do.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+---
+
+Fixed a bug in Dropdown to inoperate disabled options
```
Comment (thlmenezes) [inline .changeset/odd-nails-do.md:5]:
```suggestion
Fixes Dropdown closing after clicking a disabled option
```

I believe the description can be rewritten in simpler terms

Can you apply the suggestion?

**Review by thlmenezes — CHANGES_REQUESTED** [review]

File: .changeset/odd-nails-do.md (hunk 2 above)
Comment (mateus2033) [reply .changeset/odd-nails-do.md]: applied suggestion.

File: .changeset/odd-nails-do.md (hunk 3 above)
Comment (mateus2033) [reply .changeset/odd-nails-do.md]: applied suggestion.

**Comment by thlmenezes** [comment]
@mateus2033

- [ ] Can you downgrade the typescript-eslint package? [1]
- [ ] Can you apply prettier on the modified files? [2]

[1]

This CI error can be fixed by downgrading typescript-eslint from 8.18.1 to 8.17.0

This would fix CI for this repo

image

[2]

Running the CI lint command locally has given the following errors

```bash
$ npm run lint

> @devopness/ui-react@2.147.0 lint
> eslint --max-warnings=0 . && prettier --check .

Checking formatting...
[warn] src/components/Primitives/Dropdown/Dropdown.tsx
[warn] Code style issues found in the above file. Run Prettier with --write to fix.
```

**Comment by mateus2033** [comment]
> @mateus2033
> 
> * [x]  ~Can you downgrade the typescript-eslint package? [1]~
> * [ ]  Can you apply prettier on the modified files? [2]
> 
> [2]
> 
> This CI error can be fixed by downgrading typescript-eslint from 8.18.1 to 8.17.0, to enable local testing
> 
> This CI error will be fixed on #1105
> 
> image
> 
> Running the CI lint command locally has given the following errors
> 
> ```shell
> $ npm run lint
> 
> > @devopness/ui-react@2.147.0 lint
> > eslint --max-warnings=0 . && prettier --check .
> 
> Checking formatting...
> [warn] src/components/Primitives/Dropdown/Dropdown.tsx
> [warn] Code style issues found in the above file. Run Prettier with --write to fix.
> ```

**Comment by mateus2033** [comment]
> @mateus2033
> 
> * [x]  ~Can you downgrade the typescript-eslint package? [1]~
> * [ ]  Can you apply prettier on the modified files? [2]
> 
> [2]
> 
> This CI error can be fixed by downgrading typescript-eslint from 8.18.1 to 8.17.0, to enable local testing
> 
> This CI error will be fixed on #1105
> 
> image
> 
> Running the CI lint command locally has given the following errors
> 
> ```shell
> $ npm run lint
> 
> > @devopness/ui-react@2.147.0 lint
> > eslint --max-warnings=0 . && prettier --check .
> 
> Checking formatting...
> [warn] src/components/Primitives/Dropdown/Dropdown.tsx
> [warn] Code style issues found in the above file. Run Prettier with --write to fix.
> ```

downgrade done successfully

File: packages/ui/react/src/components/Primitives/Dropdown/Dropdown.tsx [hunk 4]
```diff
@@ -313,6 +313,11 @@ const Dropdown = ({
                         activeBackgroundColor={option.activeBackgroundColor}
                         brokenSequence={option.brokenSequence}
                         onClick={(event) => {
+
+                          if(option.isDisabled) {
```
Comment (thlmenezes) [inline packages/ui/react/src/components/Primitives/Dropdown/Dropdown.tsx:317]: @mateus2033 can you fix the prettier error on CI? Seems like running the command below should be enough

```bash
npx prettier --write packages/ui/react/src/components/Primitives/Dropdown/Dropdown.tsx
```

image

**Review by thlmenezes — CHANGES_REQUESTED** [review]

File: packages/ui/react/src/components/Primitives/Dropdown/Dropdown.tsx (hunk 4 above)
Comment (mateus2033) [reply packages/ui/react/src/components/Primitives/Dropdown/Dropdown.tsx]: Suggestion applied.

**Review by thlmenezes — APPROVED** [review]
